### PR TITLE
test: add test for ami owners and name

### PR DESCRIPTION
**Pull request by njtran**
Fixes # 

**Description**
Adds in e2e test for https://github.com/aws/karpenter/pull/3204

**How was this change tested?**

* `make apply`
* `FOCUS="ami selector" make e2etests`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [X] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/suites/integration/ami_test.go [hunk 1]
```diff
@@ -84,6 +84,49 @@ var _ = Describe("AMI", func() {
...
+		})
+		Expect(err).To(BeNil())
+		Expect(output.Images).To(HaveLen(1))
+
+		provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			AMIFamily:             &v1alpha1.AMIFamilyCustom,
+		},
+			AMISelector: map[string]string{"aws::name": *output.Images[0].Name, "aws::owners": *output.Images[0].ImageOwnerAlias},
```
Comment (tzneal) [inline test/suites/integration/ami_test.go:99]: Can you add a test without owners and ensure it fails to provision?

Comment (njtran) [reply test/suites/integration/ami_test.go:99]: Owners gets defaulted to `amazon, self`, so it should theoretically work without the `owners`. I'll add a test with mismatching owner.

Comment (tzneal) [reply test/suites/integration/ami_test.go:99]: Ah, alright, maybe a test with no owners and verify on the mock that owners is set?

**Review by bwagner5 — APPROVED** [review]
LGTM

**Pull request merged**